File: frontend/bff/src/application/orders/queries/list-orders.query.ts
import type { ApprovalStatus } from '../../../cross-cutting/enums/approval-status.js';
import type { ConfirmationStatus } from '../../../cross-cutting/enums/confirmation-status.js';
import type { PageRequest } from '../../../cross-cutting/paging/page.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface ListOrdersQuery {
  context: RequestContext;
  mandateId: string | null;
  approval: ApprovalStatus | null;
  confirmation: ConfirmationStatus | null;
  page: PageRequest;
}

File: backend/core-service/src/Fix.Application/UseCases/Orders/Queries/ListOrdersQuery.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Orders.Dtos;
using Fix.Domain.Abstractions;
using Fix.Domain.AggregateRoots.Orders;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Orders.Queries;

[RequireRole(RoleCodes.ViewOrder)]
public sealed record ListOrdersQuery(
    Guid UserId,
    Guid OrganizationId,
    Guid? MandateId,
    ApprovalStatus? Approval,
    ConfirmationStatus? Confirmation,
    int Page,
    int PageSize,
    bool WithoutMandate = false,
    bool OnlyOutside = false)
    : IQuery<PagedList<OrderDto>>, IOrganizationRequest;

